packages/api-client/src/api/serializers/common.ts

```diff
@@ -1,11 +1,13 @@
 export const filterAttachments = (attachments, type, ids) =>
-  attachments.filter((e) => e.type === type && ids.includes(e.id));
+  attachments
+    .filter((attachment) => attachment.type === type && ids.includes(attachment.id));
 
 export const extractRelationships = (attachments, type, relationship, item) => {
-  if (!item.relationships[relationship]) return [];
-  const relationships = item.relationships[relationship].data;
+  const relationships = item.relationships[relationship]?.data;
+
+  if (!relationships) return [];
 
   return Array.isArray(relationships)
     ? filterAttachments(attachments, type, relationships.map((e) => e.id))
-    : filterAttachments(attachments, type, relationships.id);
+    : filterAttachments(attachments, type, [relationships.id]);
```

Comment:
I'm looking at the code, and we don't call extractRelationships with a single item in any place ;) Maybe we could just get rid of this whole Array.isArray thing, just to keep the interface simpler.

Extracting a single relationship could have its separate function if needed, to keep the interface consistent and usable